homeassistant/components/sensor/qq.py

```diff
@@ -0,0 +1,78 @@
+"""You can use this sensor to link your qq"""
+import os
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+import threading
+
+from homeassistant.helpers.entity import Entity
+
+DOMAIN = 'qq'
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required('qq'): cv.string,
+            }),
+    },
+    extra=vol.ALLOW_EXTRA)
+
+REQUIREMENTS = ['qqbot==2.3.7']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Qqsensors."""
+    object_qq = Qqsensor()
+    add_devices([object_qq])
+    thread1 = QQ(config['qq'])
+    thread1.start()
+
+
+class Qqsensor(Entity):
+    """Representation of a Qqsensor."""
+    def __init__(self):
+        """Initialize the sensor."""
+        self._state = 'NULL'
+        self._name = DOMAIN
+        self._mutex = threading.Lock()
+
+    @property
+    def should_poll(self):
+        """need polling"""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the message of the sensor."""
+        return self._name
+
+    def update(self):
+        """get message from file"""
+        path = os.path.expanduser('~') + '/.homeassistant'
```

Comment:
This makes no sense. We already have a file sensor that can read things.

Also, you can't assume that the path is here. Use `hass.config.path`. Also don't use a filename as random as `msg.txt`, that is confusing. Also, how does QQ bot knows to write there?